File: src/main/java/org/lvtinger/paladin/profile/impl/ProfileServiceImpl.java
package org.lvtinger.paladin.profile.impl;

import org.lvtinger.paladin.profile.api.ProfileService;
import org.lvtinger.paladin.profile.api.entity.Profile;
import org.lvtinger.paladin.profile.repo.ProfileRepository;
import org.springframework.stereotype.Service;

@Service
public class ProfileServiceImpl implements ProfileService {
    private final ProfileRepository profileRepository;

    public ProfileServiceImpl(ProfileRepository profileRepository) {
        this.profileRepository = profileRepository;
    }

    @Override
    public void save(Profile profile) {
        this.profileRepository.save(profile);
    }

    @Override
    public Profile load(Long id) {
        return profileRepository.getOne(id);
    }
}

File: src/main/kotlin/org/risesun/paladin/provider/impl/ProfileServiceImpl.kt
package org.risesun.paladin.provider.impl

import org.risesun.paladin.provider.api.ProfileService
import org.risesun.paladin.provider.entity.Profile
import org.risesun.paladin.provider.repository.ProfileRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ProfileServiceImpl : ProfileService {
    @Autowired
    private lateinit var profileRepository: ProfileRepository

    override fun save(profile: Profile) {
        val exists = profileRepository.findById(profile.id).orElse(null)
        if(exists == null){
            profile.createTime = System.currentTimeMillis()
        }
        profile.updateTime = System.currentTimeMillis()
        profileRepository.save(profile)
    }

    override fun getByIds(ids: MutableCollection<Long>): MutableMap<Long, Profile> {
        val profiles = profileRepository.findAllById(ids)
        return profiles.associateBy { it.id!! }.toMutableMap()
    }
}